.travis.yml

```diff
@@ -15,13 +15,10 @@ install:
 script:
   - make test
 deploy:
-  - provider: pypi
+    provider: pypi
     distributions: sdist bdist_wheel
-    user: ons.digital
-    password:
-      secure: ms6ulxovGOxxC4JanJYv1sL/+UqrukFBBKW6TtPDHrQwQyyhHRNFkJuJeuUfPuWYYTRrK4bXsFGi6zehx6nr7+gFzL6t4a34pXAsGsqzT7/jSg5nOUst1gaE4QZSg5KUBTUBrDc939Q3Gxd89h89uxwtWV5SXqxOx0rLBmG+ZkbrsR1D/rRKI2BDxoSI1SNg+jjyIm0e9wVQg8tF19A3K1cnHFqITJh7fckQwuMgB+mD0q2H/9c/VYrpPW1u+25DPa6l/Ec0VRZKBHbQxZxXHx3Q9lnRPeBEIDWaBhw8fzi2JYkVtWlv1Npbogha8tOeB+RyEX78v58/IsKlwDJDl4bSV8EFfkb9KY6Coh+LpbpZz4i6ofotEYtfzTlyxmg8/itXySkOfayeJnE0SBG85ltxNUzcGmKusfZWyTnl1FyAk1DsY2ix3Aa8+KNHgmOBnFnCiogWULrzDinJtaHUwdWIHgDHup8DjUzPBAcR3v4xusnIFXXUuyOtmm7S0q9vvbsk//jd6a8KMcyoJ/Voytocs0ZWmNvln45VZP9Unh24tfoarhxQ2fPpY4JsgMEZobo1K9OtORVBwtT4w47NE/NWoi7WW/IPP6HKWL/iIBHJgFRhk7BMaE3KY2YtssIHzaj3U5kKKvdhpEFPPQuy5X79on1BwJ8N4/ESHsgwr8A=
+    username: __token__
+    password: $PYPI_API_TOKEN
     on:
       tags: true
-      python: 3.6
-after_success:
-  - codecov
+      python: 3.9
```

Comment:
Should we be using the minimum Python version we support here, rather than the latest? 

Also, I can't see this property in the Travis Pypi documentation at https://docs.travis-ci.com/user/deployment/pypi

Reply:
I couldn't either, but it does appear to have a functional impact. The push to PyPi only takes place on the runtime that matches the value in this key. I don't think the version used matters one bit, as we are publishing an sdist

Reply:
As this builds on multiple versions of Python, this Pypi deploy needs to be done for only one of those versions otherwise it will suffer race condition and fail for anything other than the first build the publishes to Pypi. It seemed reasonable to align this with the latest (last?) build.

Reply:
Yeah, I understand why a version has been added, but it's not documented - the documented way is to use `skip_existing: true` https://docs.travis-ci.com/user/deployment/pypi#upload-artifacts-only-once

Reply:
I think this has changed a lot since it was introduced +4 years ago and no one has been actively maintaining. I'll take a look at this approach.

Reply:
Having tested using `skip_existing` and `python` deploy approaches, my opinion is to leave it as the runtime version.

The skip_existing parameter simply tells Travis not to error the builds if Pypi returns an upload error, it does not stop the upload attempts. A race condition then occurs as one of the 4 jobs passes, with the other 3 failing to upload.

The `python` runtime param stops Travis from attempting the uploads.

Reply:
Sounds fair, it's odd that this approach isn't in their documentation.